### DISCUSSION TITLE
Is there any way to get the current scaling factor when setting fixedBondLengh in MinimalLib

### DISCUSSION THREAD

**Discussion by heberleh** [rdkit/rdkit#4877] [Q&A]
Is there any way to get the current scaling factor that would make, for example, 
all rings with 6 carbons have the same size in different images/molecules?

To make all molecules look the same, I'm using the `fixedBondLength`.

I noticed that if it doesn't fit the canvas, it is scaled.

I'm using the `MinimalLib`.

**Comment by DavidACosgrove** [comment]
Hi Henry,
One of these drawOptions will probably do the job for you, most likely the
second one.  There are a lot of drawOptions available, an ever increasing
list, that aren't especially well documented, but currently in the
$RDKIT/Code/GraphMol/MolDraw/MolDraw2D.h.  That's a C++ header, but it
should be the case that they are all exposed to the Python API as well.
Cheers,
Dave

double fixedScale =
    -1.0;  // fixes scale to this fraction of draw window width, so
           // an average bond is this fraction of the width.  If
           // scale comes out smaller than this, reduces scale, but
           // won't make it larger.  The default of -1.0 means no fix.
double fixedBondLength =
    -1.0;             // fixes the bond length (and hence the scale) to
                                            // always be this number
of pixels.  Assuming a bond
                                            // length in coordinates
is 1, as is normal.  If
                                            // scale comes out smaller
than this, reduces scale,
                                            // but won't make it
larger.  The default -1.0 means no
                                            // fix. If both fixedScale
and fixedBondLength are >
                                            // 0.0, fixedScale wins.



On Thu, Jan 13, 2022 at 11:12 AM Henry Heberle ***@***.***>
wrote:

> Is there any way to get the current scaling factor that would make, for
> example,
> all rings with 6 carbons have the same size?
>
> What I'm looking for: a canvas with good width/height is set, and only if
> a molecule is really big,
> it could be scaled to fit the width; otherwise they would have edges of
> the same type with the same width.
>
> In other words: draw with standard size, and scale if it doesn't fit the
> canvas' dimensions.
>
> Example: ring in one image is much bigger than in the other, both were
> taken from a canvas with the same width and height.
> Context: To compare molecules drawn in different scales may influence the
> analyst perception.
> [image: image]
> <https://user-images.githubusercontent.com/4249107/149319007-7dad4930-e985-486e-bfd2-be9961344430.png>
> [image: image]
> <https://user-images.githubusercontent.com/4249107/149319087-4b9d6bea-0637-42b3-b361-66fc213db2f5.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/4877>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FUL57WABQW5M7W5NZTUV2XRDANCNFSM5L3QVWYA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk
